apple/Sources/FerrostarCore/FerrostarCore.swift

```diff
@@ -252,12 +252,18 @@ public protocol FerrostarCoreDelegate: AnyObject {
     ///
     /// - Parameters:
     ///   - route: The route to navigate.
+    ///   - userLocation: The user's location. This should be as close to the users location and the start of the route
+    /// as possible. Giving fetch timing, this may not be possible, but the closer the better.
```

Comment:
Things to document:

* userLocation is an override/default, but it will try falling back to locationProvider.lastLocation
* Explain why stale locations may cause unexpected behavior.

Reply:
```suggestion
    /// as possible. If the location is too stale, the user may be almost immediately flagged as off the route,
    /// triggering a recalculation.
    /// If this parameter is `nil`, the last location will be obtained from the configured location provider automatically.
    /// If no location is available, this method will throw an exception.
```